src/main/java/org/opentripplanner/routing/algorithm/raptor/itinerary/ItineraryMapper.java

```diff
@@ -0,0 +1,333 @@
+package org.opentripplanner.routing.algorithm.raptor.itinerary;
+
+import com.conveyal.r5.otp2.api.path.AccessPathLeg;
+import com.conveyal.r5.otp2.api.path.EgressPathLeg;
+import com.conveyal.r5.otp2.api.path.Path;
+import com.conveyal.r5.otp2.api.path.PathLeg;
+import com.conveyal.r5.otp2.api.path.TransferPathLeg;
+import com.conveyal.r5.otp2.api.path.TransitPathLeg;
+import org.locationtech.jts.geom.Coordinate;
+import org.opentripplanner.api.model.Itinerary;
+import org.opentripplanner.api.model.Leg;
+import org.opentripplanner.api.model.Place;
+import org.opentripplanner.api.model.TripPlan;
+import org.opentripplanner.api.model.VertexType;
+import org.opentripplanner.common.geometry.SphericalDistanceLibrary;
+import org.opentripplanner.model.Route;
+import org.opentripplanner.model.Stop;
+import org.opentripplanner.model.Trip;
+import org.opentripplanner.model.calendar.ServiceDate;
+import org.opentripplanner.routing.algorithm.raptor.transit.Transfer;
+import org.opentripplanner.routing.algorithm.raptor.transit.TransitLayer;
+import org.opentripplanner.routing.algorithm.raptor.transit.TripSchedule;
+import org.opentripplanner.routing.core.RoutingRequest;
+import org.opentripplanner.routing.edgetype.TripPattern;
+import org.opentripplanner.routing.vertextype.TransitVertex;
+import org.opentripplanner.util.PolylineEncoder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.stream.Collectors;
+
+public class ItineraryMapper {
```

Comment:
This class definitely needs Javadoc. Ideally all classes would have Javadoc explaining what they are for, and the word "mapping" can have two different meanings. Here the mathematical idea of a function converting elements of one set into elements of another set, and the more common everyday special case of this, converting points on earth onto a flat piece of paper or screen. People approaching this code for the first time may very well assume the second meaning.